### ISSUE TITLE
Add describing comments to default exclude list entries (was: Add in-filesystem snapshots to default exclusion list)

### ISSUE THREAD

**Issue by noyannus** [bit-team/backintime#1583]
Many users run a distro that comes with a default in-filesystem snapshot, possibly with default status "activated".

Path, or even existence, of the created snapshots may not be obvious to users (dotfiles and -folders, btrfs subvolume, ...), and will clutter BiT backups, if not excluded.

Please include the most common paths to such backups in the default exclusion list.

**Comment by emtiu** [comment]
Are you sure that btrfs subvolumes are picked up by rsync and land in Back In Time backups? Have you tested this, and could you provide some examples of paths that should be excluded?

**Comment by noyannus** [comment]
> Are you sure that btrfs subvolumes are picked up by rsync and land in Back In Time backups? 

Unfortunately not. I have hopped distro a while ago from one creating subvolumes with TimeShift to one that places snapshots under `/` (Snapper).

I'd recommend for the SUSE distro family, and likely for everyone using Snapper to exclude ` */.snapshots/*`. Snapper defaults to place snapshots in root of the snapshotted subvolume, e.g. `/.snapshots` `/home/.snapshots`. It can use subvolumes mounted on them, but they would look the same to BiT.

**Comment by buhtz** [comment]
What is Snapper? Can you give us a link please?

I would say that `*/.snapshots/*` is to broad (sensitive) to make it a default exclude.

**Comment by noyannus** [comment]
Snapper is a btrfs snapshotting tool, somewhat like Timeshift. 

It is not widely known; probably because [it originates from a German company](https://documentation.suse.com/sles/12-SP5/html/SLES-all/cha-snapper.html) whose distros are underrepresented in the anglophone Linux world. 

A good start info is the [Arch Wiki](https://wiki.archlinux.org/title/snapper). With a [GUI](https://gitlab.com/btrfs-assistant/btrfs-assistant), Snapper is really neat.

A search for 'timeshift vs snapper' brings up useful comparisons (or so my notes from some years ago say...)

**Comment by aryoda** [comment]
Snapper source code: https://github.com/openSUSE/snapper

Ubuntu provides the packages `snapper` and `snapper-gui` so Debian may have these packages too...

**Comment by aryoda** [comment]
I think the path `.snapshot` must be excluded and is by default located under the root folder.

But since `snapper` can store snapshots in "[subvolumes](https://btrfs.readthedocs.io/en/latest/Subvolumes.html)" (I have no idea if this term means the same for all file systems) it may be more complicated.

BiT is not meant to make backups of a system (OS) root folder but I guess the `/.snapshot` folder could also appear in a mounted pure data partition.

Do we know examples (apps) where excluding `/.snapshot` would lead to a problem because it contains real data and should be included in BiT backups?

**Comment by noyannus** [comment]
Snapshots of `/home` partitions are [ not unheard of](https://en.opensuse.org/openSUSE:Snapper_Tutorial#Adding_/home_To_Snapper).

A quick search for `application "~/.snapshots/"` finds no page with both. I think it is reasonably safe to exclude `~/.snapshots/`, but should be accompanied with a brief warning in the readme.

**Comment by buhtz** [comment]
> Ubuntu provides the packages `snapper` and `snapper-gui` so Debian may have these packages too...

Of course it is [in Debian](https://tracker.debian.org/pkg/snapper). Ubuntu/Canonical would never work them self. They only take the work of others and put their own label on it. 😆 😈 

I see no problem with excluding `~/.snapshots/` and `/.snapshots/` by default. It is different from the initial proposal `*/.snapshots/*`.

**Comment by noyannus** [comment]
> I see no problem with excluding ~/.snapshots/ and /.snapshots/ by default. It is different from the initial proposal */.snapshots/*.

Yeah, I had not thougth through the ramifications. Feel free to close the issue.

**Comment by emtiu** [comment]
I think the default exclude list for all BiT users doesn't need to be changed.

btrfs is not very common (yet), and tools like Snapper even less so.  If a user has both Snapper and Back In Time running, they probably understand the implications, and can adjust the exclude list for themselves.

But my opinion is not very strong, and a change to the default exclude list is not "expensive" (for us as maintainers) :)

**Comment by noyannus** [comment]
btrfs has matured and gained stability in recent years. It's hard to find reliable data for and popularity. One data point, [this poll in the Endeavour forum](https://forum.endeavouros.com/t/poll-which-file-system-in-2022-ext4-btrfs-xfs-zfs-f2fs-choose-1/26207) from 1 1/2 years ago, shows it nearly as liked as ext4 (43% vs 49%; and ext4 means something!). Outside Endeavour, with distros for an on average less technically inclined user base, it may be somewhat less, though. 

Snapper is more obscure, but seems to be on the rise as a Timeshift alternative with overlapping but in parts different functionality.

**Comment by emtiu** [comment]
Thanks, that's interesting to know :)

**Comment by noyannus** [comment]
Your're welcome. :-)

**Comment by aryoda** [comment]
I think our growing exclusion list demands for providing comments to the users (and us) in the GUI to explain why a pattern is in the exclusion list (eg. via tool tips or a second "Notes" column). Otherwise it is difficult to understand the impact of (not) excluding some folders.

![image](https://github.com/bit-team/backintime/assets/11374410/3e8d594d-4308-49a9-992f-50a69a7c1848)

Is providing comments for exclusion list entries worth a new feature request issue?

**Comment by aryoda** [comment]
> I see no problem with excluding `~/.snapshots/` and `/.snapshots/` by default.

Can we agree on adding this to our default exclusion list? Any objections?

**Edit:** This could be done together with adding mozilla and discord files to the exclusion list (https://github.com/bit-team/backintime/issues/1555#issuecomment-1788228360)

**Comment by buhtz** [comment]
OK, for me.

**Comment by noyannus** [comment]
> I think our growing exclusion list demands for providing comments to the users (and us) in the GUI to explain why a pattern is in the exclusion list (eg. via tool tips or a second "Notes" column). Otherwise it is difficult to understand the impact of (not) excluding some folders.

Just an idea: The exclusion patterns list has enough room for a second column with a very brief explanation, e.g.
```

/tmp/*         Temporary files, deleted at every boot
/mysecrets/    Must never leave this machine

```
BiT could then provide a default text in a user editable text field (table cell), so users can add their own notes to the defaults and annotate their own exclusions. With this exclusion table made horizontally scrollable, the notes could be arbitrarily long.

**Comment by Germar** [comment]
> I think the default exclude list for all BiT users doesn't need to be changed.

I'd second this. The default exclude is just for inexperienced users running backups of their home folder. Everyone else need to research on his/her own what need to exclude. Especially when it comes to backups of FS_ROOT.

Adding a comment line for each in-/exclude is a good idea. And if you already touch that part maybe add a checkbox to en-/disable each line.

**Comment by buhtz** [comment]
I make it short. Closing the issue as not planed. Wouldn't add these "snapshot" dirs to the exclude list (default or only as suggestion).

But thank you for the discussion and the insights.

Please note that BIT now has the _Exclude Suggestions_ dialog. See #2309. But even here, I wouldn't add the "snapshot" dir.

![Peek 2025-12-13 13-45](https://github.com/user-attachments/assets/d8c3be85-7ba0-44cb-8767-8993752e986b)
